modules/Bio/EnsEMBL/Variation/DBSQL/LDFeatureContainerAdaptor.pm

```diff
@@ -423,15 +421,29 @@ sub _fetch_by_Slice_VCF {
         my $vcf_file = $vc->_get_vcf_filename_by_chr($slice->seq_region_name);
         throw("ERROR: Can't get VCF file\n") unless $vcf_file;
         push @files, $vcf_file;
-        my $loc_string = sprintf("%s:%i-%i", $slice->seq_region_name, $slice->start, $slice->end);
+        my $chr = $slice->seq_region_name;
+        if ($vc->use_seq_region_synonyms) {
+          my @chr_in_vcf_file = @{$vc->_vcf_parser_obj($vcf_file)->{tabix_file}->seqnames};
+          my @synonyms = ();
+          foreach my $synonym (@{$vc->_get_synonyms_by_chr($chr)}) {
+            push @synonyms, $synonym if (grep {$_ eq $synonym} @chr_in_vcf_file);
+          }
+          $chr = $synonyms[0];
+          warn "use_seq_region_synonyms is set. Found more than one synonym for sequence name $chr" if (scalar @synonyms > 1);
```

Comment:
Suggest putting checks of @synonyms in an if/elsif loop. If @synonyms > 1, do no need to check if no synonyms.

Reply:
I didn't use if/elsif because I don't have anything for the else clause to report. I'm also testing for no synonyms because use_seq_region_synonyms is set which means that if there wasn't a synonym found for one of the chromosomes in the VCF file it would be worth finding out why we don't have a synonym for it.

Reply:
An else would not be needed, code would continue. Was just noting if  @synonyms > 1, then no need to check scalar @synonyms == 0. 
if (scalar @synonyms > 1) {
  warn "use_seq_region_synonyms is set. Found more than one synonym for sequence name $chr";
} elsif (! @synonyms) {
  warn "use_seq_region_synonyms is set. But didn't find synonym for sequence name $chr";
}

Reply:
That makes sense. I will update the PR with your suggestion.